scripts/irods/test/test_all_rules.py

```diff
@@ -1818,6 +1819,41 @@ def test_msiSetKeyValuePairsToObj_does_not_crash__issue_7027(self):
 
         self.admin.assert_icommand(['imeta', 'ls', '-d', TEST_FILE], 'STDOUT', [ 'attribute: key\nvalue: val\n' ])
 
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-python', "Requires the PREP.")
+    def test_msiDataObjChksum_does_not_lead_to_segfault_on_good_verification_result__issue_7859(self):
+        with temporary_core_file() as core:
+            attr_key = 'issue_7859'
+            expected_attr_units = 'issue_7859 is working'
+
+            # Add a rule that attempts to verify the checksum using msiDataObjChksum.
+            # If verification finds no issues, the checksum output variable will hold an empty string.
+            # The check for the empty string and attachment of metadata give the test a way to determine
+            # the outcome of the rule.
+            core.add_rule(textwrap.dedent(f'''
+            def pep_api_data_obj_trim_pre(rule_args, callback, rei):
+                logical_path = str(rule_args[2].objPath)
+                result = callback.msiDataObjChksum(logical_path, 'verifyChksum=', '')
+                if result['arguments'][2] == '':
+                    callback.msiModAVUMetadata('-d', logical_path, 'set', '{attr_key}', '{attr_key}', '{expected_attr_units}')
+            '''))
+
+            # Create a new data object.
+            data_object = 'issue_7859_data_object'
+            self.user0.assert_icommand(['itouch', data_object])
+
+            # Calculate a checksum so the rule has information it can use for verification.
+            # Without this, the rule will fail.
+            self.user0.assert_icommand(['ichksum', data_object], 'STDOUT', ['sha2:47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='])
+
+            # Trigger the PEP.
+            # Before the fix, this would result in an error.
+            self.user0.assert_icommand(['itrim', data_object], 'STDOUT', ['Number of files trimmed = 0.'])
+
+            # Show the checksum verification was successful by searching for the expected attribute units.
+            self.user0.assert_icommand(
+                ['iquest', f"select META_DATA_ATTR_UNITS where META_DATA_ATTR_NAME = '{attr_key}' and META_DATA_ATTR_VALUE = '{attr_key}'"],
+                'STDOUT',
+                [f'META_DATA_ATTR_UNITS = {expected_attr_units}\n'])
```

Comment:
The following suggestion is completely optional. :)

There is a function called `metadata_attr_with_value_exists` in `lib.py` which does something *almost* like this, and I think could act as a valid replacement for these lines. The AVU being annotated in the rule for this test is using the same attribute and value string, and a unique-ish string in the units field. If we put the unique-ish string in the value field instead and drop the units, we could use `metadata_attr_with_value_exists` here, which selects the `META_DATA_ATTR_VALUE` instead of `META_DATA_ATTR_UNITS`. ...unless there's a reason for using the units field.

Reply:
I originally used that function, but decided to go in another direction because it doesn't include the target data object. AFAIK, that function only checks that the AVU exists in the database. It doesn't assert that the AVU is attached to the data object.

I plan to open an issue for that today.

Reply:
Ah, I noticed that as well. However, this is also not asserting that the metadata is attached to the data object, though, so perhaps we should do that here?

Reply:
You're right lol. Definitely need to include the data object.

Reply:
Fixed.